core/src/Cuda/Kokkos_CudaSpace.cpp

```diff
@@ -229,7 +229,7 @@ void *CudaSpace::impl_allocate(
 
 #ifndef CUDART_VERSION
 #error CUDART_VERSION undefined!
-#elif (CUDART_VERSION >= 11020)
+#elif (CUDAMALLOCASYNC && CUDART_VERSION >= 11020)
```

Comment:
```suggestion
#elif (KOKKOS_ENABLE_CUDA_MALLOC_ASYNC && CUDART_VERSION >= 11020)
```

Reply:
Making one change to spelling as it seems to error out of build if Kokkos isn't in Camel case

Reply:
@masterleinad Build is getting this error here
`/home/mstack/0_Kokkos/Kokkos_dev/core/src/Cuda/Kokkos_CudaSpace.cpp:232:40: error: operator '&&' has no left operand
 #elif (KOKKOS_ENABLE_CUDA_MALLOC_ASYNC && CUDART_VERSION >= 11020)`

Reply:
Are we sure we dont need the extra preprocessor flag from add_compile_definitions?

Reply:
Resolved from Slack discussion with Daniel, needed to add defined() around the option